### PR TITLE
Change GetSignedAssertion public API

### PR DESCRIPTION
**Pull request by bgavrilMS**
# Change GetSignedAssertion public API

Fix for #2853 2853

- [ ] You've read the [Contributor Guide](https://github.com/AzureAD/microsoft-identity-web/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/AzureAD/microsoft-identity-web/blob/master/CODE_OF_CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

{Detail}

Fixes #{bug number} (in this specific format)

### CODE REVIEW

File: src/Microsoft.Identity.Web.Certificateless/AzureIdentityForKubernetesClientAssertion.cs [hunk 1]
```diff
@@ -65,7 +66,7 @@ public AzureIdentityForKubernetesClientAssertion(string? filePath, ILogger? logg
         /// Get the signed assertion from a file.
         /// </summary>
         /// <returns>The signed assertion.</returns>
-        protected override Task<ClientAssertion> GetClientAssertion(CancellationToken cancellationToken)
+        protected override Task<ClientAssertion> GetClientAssertion(AssertionRequestOptions assertionRequestOptions)
```
Comment (westin-m) [inline src/Microsoft.Identity.Web.Certificateless/AzureIdentityForKubernetesClientAssertion.cs:69]: Can you address the github warnings around this?

File: src/Microsoft.Identity.Web.Certificate/SignedAssertionFilePathCredentialsLoader.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public async Task LoadIfNeededAsync(CredentialDescription credentialDescription,
                 {
                     // Given that managed identity can be not available locally, we need to try to get a
                     // signed assertion, and if it fails, move to the next credentials
-                    _= await signedAssertion!.GetSignedAssertion(CancellationToken.None);
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.Certificate/SignedAssertionFilePathCredentialsLoader.cs:39]: An Async method is supposed to end with Async and have a cancellation token as the last parameter? Maybe we should do that?

Comment (jmprieur) [reply src/Microsoft.Identity.Web.Certificate/SignedAssertionFilePathCredentialsLoader.cs:39]: oh I see the cancellation token is in the AssertionRequestOptions ... a bit strange, but works

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @bgavrilMS

File: src/Microsoft.Identity.Web.Certificate/SignedAssertionFilePathCredentialsLoader.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/Microsoft.Identity.Web.Certificate/SignedAssertionFilePathCredentialsLoader.cs:39]: Agreed on the "Async" suffix.

It's a fair comment for Options object + CancellationToken. For keeping things simple with MSAL.NET, let's leave it as is ?

**Pull request merged**